src/components/Layouts/Primary/TopBar/Items/AddMenu/AddMenu.tsx

```diff
@@ -1,53 +1,171 @@
+  const { state, openTaskDrawer } = useApp();
+  const { accountListId } = state;
+  const [selectedMenuItem, changeSelectedMenuItem] = useState(-1);
+  const [dialogOpen, changeDialogOpen] = useState(false);
+
+  const handleAddTaskClick = (): void => {
+    setAnchorEl(undefined);
+    openTaskDrawer({});
+  };
+
+  const handleMenuItemClick = (menuItem: number) => {
+    changeSelectedMenuItem(menuItem);
+    changeDialogOpen(true);
+  };
+
+  const handleDialogClose = () => {
+    changeDialogOpen(false);
+  };
+
+  const renderDialog = () => {
+    const renderDialogContent = () => {
+      switch (selectedMenuItem) {
+        case 0:
+          return (
+            <CreateContact
+              accountListId={accountListId ?? ''}
+              handleClose={handleDialogClose}
+            />
+          );
+      }
+    };
+    return (
+      <Dialog
+        open={dialogOpen}
+        aria-labelledby={t('Create Contact Dialog')}
+        fullWidth
+        maxWidth="sm"
+      >
+        {renderDialogContent()}
+      </Dialog>
+    );
+  };
+
+  const addMenuContent = [
+    {
+      text: 'Add Contact',
+      icon: <PersonIcon />,
+    },
+    {
+      text: 'Multiple Contacts',
+      icon: <PeopleIcon />,
+      customOnClick: () => console.log('multiple contact'),
+    },
+    {
+      text: 'Add Donation',
+      icon: <CardGiftcardIcon />,
+      customOnClick: () => console.log('add donation'),
+    },
+    {
+      text: 'Add Task',
+      icon: <ListIcon />,
+      customOnClick: handleAddTaskClick,
+    },
+    {
+      text: 'Log Task',
+      icon: <EditIcon />,
+      customOnClick: () => console.log('log task'),
+    },
+  ];
 
   const [anchorEl, setAnchorEl] = useState<EventTarget & HTMLButtonElement>();
 
   return (
     <>
       <IconButton
-        className={classes.link}
         aria-controls="add-menu"
         aria-haspopup="true"
         onClick={(event) => setAnchorEl(event.currentTarget)}
       >
-        <AddIcon />
+        <HoverAddIcon titleAccess="Add Button" />
       </IconButton>
-      <Menu
+      <MenuContainer
         id="add-menu"
-        anchorEl={anchorEl}
         keepMounted
         open={Boolean(anchorEl)}
         onClose={() => setAnchorEl(undefined)}
-        classes={{ paper: classes.menuPaper }}
+        anchorEl={anchorEl}
+        getContentAnchorEl={null}
+        anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
+        transformOrigin={{ vertical: 'top', horizontal: 'center' }}
       >
-        <Box display="flex" flexDirection="row" justifyContent="center">
-          <Box flexGrow={1}>{t('Add Menu')}</Box>
+        <Box display="flex" flexDirection="column" justifyContent="center">
+          {addMenuContent.map(({ text, icon, customOnClick }, index) => (
+            <RowContainer
+              key={index}
+              display="flex"
+              onClick={() =>
+                customOnClick ? customOnClick() : handleMenuItemClick(index)
```

Comment:
Seems a little odd to have 2 ways of doing this.

Reply:
Seems like it might be nicest located with the menu item definition instead of having to deal with indexes.
```
    {
      text: 'Add Contact',
      icon: <PersonIcon />,
      onClick: () => console.log('todo'),
    }
```
Unless there's some case that can't handle.

Reply:
Ah I like that better, I'll do that 🙂